### PR TITLE
LLREST: Add PreferHasAttributeNodeSelector

### PR DESCRIPTION
**Pull request by il-kyun**
Add PreferHasAttributeNodeSelector.
PreferHasAttributeNodeSelector and HasAttributeNodeSelector will work the same if there is a node with particular attribute in the attributes, but PreferHasAttributeNodeSelector will select another node even if there is no node with particular attribute in the attributes.

### CODE REVIEW

File: client/rest/src/test/java/org/elasticsearch/client/PreferHasAttributeNodeSelectorTests.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client;
+
+import org.apache.http.HttpHost;
+import org.elasticsearch.client.Node.Roles;
+
+import java.util.*;
```
Comment (nik9000) [inline client/rest/src/test/java/org/elasticsearch/client/PreferHasAttributeNodeSelectorTests.java]: We fail the build on `*` imports actually.

File: client/rest/src/main/java/org/elasticsearch/client/PreferHasAttributeNodeSelector.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+package org.elasticsearch.client;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A {@link NodeSelector} that prefer to select nodes that have a particular value
+ * for an attribute.
```
Comment (nik9000) [inline client/rest/src/main/java/org/elasticsearch/client/PreferHasAttributeNodeSelector.java]: I think it is important to compare this to `HasAttributeNodeSelector` in the javadoc explicitly.

**Review by nik9000 — COMMENTED** [review]
I think this is a useful addition! I think it'd be nice to explicitly compare it to `HasAttributeNodeSelector` in its javadoc and in the PR description. The PR description will eventually become the commit message so it is nice to explain what is up.

File: client/rest/src/test/java/org/elasticsearch/client/PreferHasAttributeNodeSelectorTests.java (hunk 1 above)
Comment (il-kyun) [reply client/rest/src/test/java/org/elasticsearch/client/PreferHasAttributeNodeSelectorTests.java]: sorry, I will fix.

File: client/rest/src/main/java/org/elasticsearch/client/PreferHasAttributeNodeSelector.java (hunk 2 above)
Comment (il-kyun) [reply client/rest/src/main/java/org/elasticsearch/client/PreferHasAttributeNodeSelector.java]: Thanks for your advice.
I added javadoc.

**Comment by nik9000** [comment]
@elasticmachine, test this please.

**Review by nik9000 — APPROVED** [review]
The implementation looks good to me and I think this is a good thing to add.

**Comment by nik9000** [comment]
@il-kyun, I've merged this and backported it to 6.x. Thanks!

**Pull request merged**